src/app/(proper_react)/redesign/(authenticated)/user/dashboard/fix/leaked-passwords/LeakedPasswordsLayout.tsx

```diff
@@ -8,51 +8,86 @@ import { ResolutionContainer } from "../ResolutionContainer";
 import { ResolutionContent } from "../ResolutionContent";
 import { Button } from "../../../../../../../components/server/Button";
 import { useL10n } from "../../../../../../../hooks/l10n";
-import { LeakedPassword } from "./leakedPasswordsData";
+import {
+  LeakedPasswordsTypes,
+  getLeakedPasswords,
+} from "./leakedPasswordsData";
 import Link from "next/link";
 import { getLocale } from "../../../../../../../functions/universal/getLocale";
+import {
+  StepDeterminationData,
+  StepLink,
+  getNextGuidedStep,
+} from "../../../../../../../functions/server/getRelevantGuidedSteps";
+import { FixView } from "../FixView";
+import { getGuidedExperienceBreaches } from "../../../../../../../functions/universal/guidedExperienceBreaches";
 
 export interface LeakedPasswordsLayoutProps {
-  label: string;
-  pageData: LeakedPassword;
+  type: LeakedPasswordsTypes;
+  subscriberEmails: string[];
+  data: StepDeterminationData;
 }
 
-export function LeakedPasswordsLayout({
-  pageData,
-}: LeakedPasswordsLayoutProps) {
+export function LeakedPasswordsLayout(props: LeakedPasswordsLayoutProps) {
   const l10n = useL10n();
-  const { title, illustration, content } = pageData;
+
+  const stepMap: Record<LeakedPasswordsTypes, StepLink["id"]> = {
+    password: "LeakedPasswordsPassword",
+    "security-question": "LeakedPasswordsSecurityQuestion",
+  };
+
+  const guidedExperienceBreaches = getGuidedExperienceBreaches(
+    props.data.subscriberBreaches,
+    props.subscriberEmails
+  );
+
+  const pageData = getLeakedPasswords({
+    dataType: props.type,
+    breaches: guidedExperienceBreaches,
+  });
+
+  // The non-null assertion here should be safe since we already did this check
+  // in `./[type]/page.tsx`:
+  const { title, illustration, content } = pageData!;
 
   return (
-    <ResolutionContainer
-      type="leakedPasswords"
-      title={title}
-      illustration={illustration}
-      cta={
-        <>
-          <Button
-            variant="primary"
-            small
-            // TODO: Add test once MNTOR-1700 logic is added
-            /* c8 ignore next 3 */
-            onPress={() => {
-              // TODO: MNTOR-1700 Add routing logic
-            }}
-            autoFocus={true}
-          >
-            {l10n.getString("leaked-passwords-mark-as-fixed")}
-          </Button>
-          <Link
-            // TODO: Add test once MNTOR-1700 logic is added
-            href="/"
-          >
-            {l10n.getString("leaked-passwords-skip")}
-          </Link>
-        </>
+    <FixView
+      subscriberEmails={props.subscriberEmails}
+      data={props.data}
+      nextStepHref={
+        getNextGuidedStep(props.data, stepMap[props.type])?.href ?? ""
       }
-      estimatedTime={4}
     >
-      <ResolutionContent content={content} locale={getLocale(l10n)} />
-    </ResolutionContainer>
+      <ResolutionContainer
```

Comment:
All the content in here is the same except for additional indenting, but now wrapped in `<FixView>`, and with the data passed in as props.